cli/src/upload.rs

```diff
@@ -72,10 +72,19 @@ pub struct UploadArgs {
     pub team: Option<String>,
     #[arg(long, help = "Value to override CODEOWNERS file or directory path.")]
     pub codeowners_path: Option<String>,
-    #[arg(long, help = "Run commands with the quarantining step.")]
+    #[arg(
+        long,
+        help = "Run commands with the quarantining step.",
+        default_value = "true"
+    )]
     pub use_quarantining: bool,
-    #[arg(long, help = "Do not fail if no junit files are found.")]
-    pub allow_missing_junit_files: bool,
+    #[arg(
+        long,
+        alias = "allow-missing-junit-files",
```

Comment:
Adds this deprecated flag as a hidden alias:

<details>
<summary>

Click to view `upload` command `--help` text

</summary>

```text
Upload data to Trunk Flaky Tests

Usage: trunk flakytests upload [OPTIONS] --org-url-slug <ORG_URL_SLUG> --token <TOKEN>

Options:
      --junit-paths <JUNIT_PATHS>
          Comma-separated list of glob paths to junit files.
      --xcresult-path <XCRESULT_PATH>
          Path of xcresult directory
      --org-url-slug <ORG_URL_SLUG>
          Organization url slug.
      --token <TOKEN>
          Organization token. Defaults to TRUNK_API_TOKEN env var. [env: TRUNK_API_TOKEN=]
      --repo-root <REPO_ROOT>
          Path to repository root. Defaults to current directory.
      --repo-url <REPO_URL>
          Value to override URL of repository.
      --repo-head-sha <REPO_HEAD_SHA>
          Value to override SHA of repository head.
      --repo-head-branch <REPO_HEAD_BRANCH>
          Value to override branch of repository head.
      --repo-head-commit-epoch <REPO_HEAD_COMMIT_EPOCH>
          Value to override commit epoch of repository head.
      --tags <TAGS>
          Comma separated list of custom tag=value pairs.
      --print-files
          Print files which will be uploaded to stdout.
      --dry-run
          Run metrics CLI without uploading to API.
      --team <TEAM>
          Value to tag team owner of upload.
      --codeowners-path <CODEOWNERS_PATH>
          Value to override CODEOWNERS file or directory path.
      --use-quarantining
          Run commands with the quarantining step.
      --allow-empty-test-results
          Do not fail if test results are not found.
  -h, --help
          Print help
```

</details>